src/rebar3_hank_prv.erl

```diff
@@ -120,6 +127,80 @@ format_result(#{file := File,
                 text := Msg}) ->
     hank_utils:format_text("~ts:~tp: ~ts", [File, Line, Msg]).
 
+-spec maybe_write_data_to_json_file([hank_rule:result()], rebar_state:t()) -> ok.
+maybe_write_data_to_json_file(Result, State) ->
+    {Args, _} = rebar_state:command_parsed_args(State),
+    case lists:keyfind(output_json_file, 1, Args) of
+        {output_json_file, JsonFilePath} ->
+            case valid_json_format(JsonFilePath) of
+                true ->
+                    ConvertedResult = convert_data_to_binary(Result),
+                    EncodedResult = jsx:encode(ConvertedResult),
+                    ok = file:write_file(JsonFilePath, EncodedResult);
+                false ->
+                    ok
+            end;
+        _ ->
+            ok
+    end.
+
+-spec valid_json_format(string()) -> boolean().
+valid_json_format(JsonFilePath) ->
+    JsonFileName = lists:last(string:tokens(JsonFilePath, "/")),
+    case lists:last(string:tokens(JsonFileName, ".")) of
+        "json" ->
+            true;
+        _ ->
+            false
+    end.
+
+-spec convert_data_to_binary([hank_rule:result()]) -> list().
+convert_data_to_binary(Data) ->
+    Func =
+        fun(RuleDetailMap) ->
+            #{file := FileName, line := Line, rule := RuleBroken, text := Description} = RuleDetailMap,
+            #{
+                <<"path">> => to_binary(FileName),
+                <<"start_line">> => Line,
+                <<"hank_rule_broken">> => to_binary(RuleBroken),
+                <<"title">> => compute_title(RuleBroken),
+                <<"message">> => to_binary(Description)}
+        end,
+    [Func(RuleDetails) || RuleDetails <- Data].
```

Comment:
Instead of adding a function here, I would put this into `hank_rule:result_to_json/1` and I'll just call this here like this…
```suggestion
-spec convert_data_to_binary([hank_rule:result()]) -> list().
convert_data_to_binary(Results) ->
    lists:map(fun hank_rule:result_to_json/1, Results).
```

Reply:
Done.